nova_vm/src/ecmascript/builtins/control_abstraction_objects/promise_objects/promise_abstract_operations/promise_jobs.rs

```diff
@@ -0,0 +1,160 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## [27.2.2 Promise Jobs](https://tc39.es/ecma262/#sec-promise-jobs)
+
+use crate::{ecmascript::{
+    abstract_operations::operations_on_objects::{call_function, get_function_realm}, builtins::{control_abstraction_objects::promise_objects::promise_abstract_operations::promise_capability_records::PromiseCapability, promise::Promise, ArgumentsList}, execution::{agent::{InnerJob, Job, JsError}, Agent, JsResult}, types::{Function, IntoValue, Object, Value}
+}, heap::CreateHeapData};
+
+use super::{
+    promise_reaction_records::{PromiseReaction, PromiseReactionHandler, PromiseReactionType},
+    promise_resolving_functions::{PromiseResolvingFunctionHeapData, PromiseResolvingFunctionType},
+};
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct PromiseResolveThenableJob {
+    promise_to_resolve: Promise,
+    thenable: Object,
+    then: Function,
+}
+impl PromiseResolveThenableJob {
+    pub(crate) fn run(self, agent: &mut Agent) -> JsResult<()> {
```

Comment:
**suggestion:** Preferably add a comment about what part of the spec this corresponds to.